docs/api-reference/aggregation-layers/aggregator.md

```diff
@@ -0,0 +1,168 @@
+# Aggregator Interface
+
+The `Aggregator` interface describes a type of class that performs aggregation.
+
+## Terminology
+
+_Aggregation_ is a 2-step process:
+
+1. **Sort**: Group a collection of _data points_ by some property into bins_.
+2. **Aggregate**: for each _bin_, calculate a numeric output (_result_) from some metrics (_values_) from all its members. Multiple results can be obtained independently (_channels_).
+
+An implementation of the _Aggregator_ interface takes the following inputs:
+- The number of data points
+- The group that each data point belongs to, by mapping each data point to a _binId_ (array of integers)
+- The values to aggregate, by mapping each data point in each channel to one _value_ (number)
+- The method (_operation_) to reduce a list of values to one number, such as SUM
+
+And yields the following outputs:
+- A list of _binIds_ that data points get sorted into
+- The aggregated values (_result_) as a list of numbers, comprised of one number per bin per channel
+- The [min, max] among all aggregated values (_domain_) for each channel
+
+### Example
+
+Consider the task of making a [histogram](https://en.wikipedia.org/wiki/Histogram) that shows the result of a survey by age distribution.
+
+1. The _data points_ are the list of participants, and we know the age of each person.
+2. Suppose we want to group them by 5-year intervals. A 21-year-old participant is assigned to the bin of age 20-25, with _binId_ `[20]`. A 35-year-old participant is assigned to the bin of age 35-40, with _binId_ `[35]`, and so on.
+3. For each bin (i.e. age group), we calculate 2 _values_:
+    + The first _channel_ is "number of participants". Each participant in this group yields a _value_ of 1, and the result equals all values added together (_operation_: SUM).
+    + The second _channel_ is "average score". Each participant in this group yields a _value_ that is their test score, and the result equals the sum of all scores divided by the number of participants (_operation_: MEAN).
+4. As the outcome of the aggregation, we have:
+    + Bins: `[15, 20, 25, 30, 35, 40]`
+    + Channel 0 result: `[1, 5, 12, 10, 8, 3]`
+    + Channel 0 domain: `[1, 12]`
+    + Channel 1 result: `[6, 8.2, 8.5, 7.9, 7.75, 8]`
+    + Channel 1 domain: `[6, 8.5]`
+
+
+## Methods
+
+An implementation of `Aggregator` should expose the following methods:
+
+#### `setProps` {#setprops}
+
+Set runtime properties of the aggregation.
+
+```ts
+aggregator.setProps({
+  pointCount: 10000,
+  attributes: {...},
```

Comment:
One day we should call this `columns`. Not sure if we could go that far now....

Reply:
These are the same format as `props.data.attributes` and we have a bunch of classes/types along that line such as `AttributeManager`, `BinaryAttribute` etc. At least for now I don't think we should add more concepts to the mix.

Reply:
Makes sense. But I would love to give this an overhaul in deck.gl v10 as part of deep Arrow integration etc.